src/dlgRoomProperties.h

```diff
@@ -0,0 +1,91 @@
+ *   Copyright (C) 2022 by Stephen Lyons - slysven@virginmedia.com         *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
+ ***************************************************************************/
+
+#include "Host.h"
+
+#include "pre_guard.h"
+#include "ui_room_properties.h"
+#include "post_guard.h"
+
+
+class dlgRoomProperties : public QDialog, public Ui::room_properties
+{
+    Q_OBJECT
+
+public:
+    Q_DISABLE_COPY(dlgRoomProperties)
+    explicit dlgRoomProperties(Host*, QWidget* parent = nullptr);
+    void init(
+        QHash<QString, int> usedNames,
+        QHash<int, int>& pColors,
+        QHash<QString, int>& pSymbols,
+        QHash<int, int>& pWeights,
+        QHash<bool, int> lockStatus,
+        QSet<TRoom*>& pRooms);
+    void accept() override;
+
+signals:
+    void signal_save_symbol(
+        bool changeName, QString newName,
+        bool mChangeRoomColor, int mRoomColorNumber,
+        bool changeSymbol, QString newSymbol,
+        bool changeSymbolColor, QColor newSymbolColor,
+        bool changeWeight, int newWeight,
+        bool changeLockStatus, bool newLockStatus,
+        QSet<TRoom*> mpRooms);
+
+private:
+    QColor backgroundBasedColor(QColor);
+    QColor defaultSymbolColor();
+    QString getNewSymbol();
+    void initSymbolInstructions();
+    QStringList getComboBoxSymbolItems();
+    QFont getFontForPreview(QString);
+
+    int getNewWeight();
+    void initWeightInstructions();
+    QStringList getComboBoxWeightItems();
+    void initLockInstructions();
+
+    Host* mpHost;
+    QSet<TRoom*> mpRooms;
+    QHash<QString, int> mpSymbols;
+    QHash<int, int> mpWeights;
+    QColor selectedSymbolColor;
+    QColor mRoomColor;
+    int mRoomColorNumber;
+    bool mChangeRoomColor = false;;
+    QString multipleValuesPlaceholder = tr("(Multiple values...)");
+
```

Comment:
There are a couple of members without initialisers here, and a bogus extra semicolon:
```diff
-    Host* mpHost;
+    Host* mpHost = nullptr;
     QSet<TRoom*> mpRooms;
     QHash<QString, int> mpSymbols;
     QHash<int, int> mpWeights;
     QColor selectedSymbolColor;
     QColor mRoomColor;
-    int mRoomColorNumber;
+    int mRoomColorNumber = -1;
-   bool mChangeRoomColor = false;;
+   bool mChangeRoomColor = false;
     QString multipleValuesPlaceholder = tr("(Multiple values...)");

```

Reply:
Agree on the bogus semicolon.
Didn't know about necessary initializations.
Searching for `Host* mpHost;` yields 4 results, whereas `Host* mpHost = nullptr;` is found 3 times.

Reply:
I haven't :blush: finished going through the class initialisation lists #5890 and #5937 have only done some classes...